src/bloqade/pyqrack/task.py

```diff
@@ -51,3 +55,80 @@ def qubits(self) -> list[PyQrackQubit]:
         except AttributeError:
             Warning("Task has not been run, there are no qubits!")
             return []
+
+    def multirun(self, shots: int = 1) -> dict[RetType, float]:
+        """
+        Repeatedly run the task to collect statistics on the shot outcomes
+          The average is done over nshots and thus is frequentist and converges to
+          exact only in the shots -> infinity limit.
+        Parameters:
+        shots - the number of repetitions of the task
+        Returns:
+        dict[RetType, float] - a dictionary mapping outcomes to their probabilities,
+          as estimated from counting the shot outcomes. RetType must be hashable.
+        """
+
+        results: list[RetType] = [self.run() for _ in range(shots)]
+
+        # Convert IList to tuple so that it is hashable by Counter
+        def convert(data):
+            if isinstance(data, (list, IList)):
+                return tuple(convert(item) for item in data)
+            return data
+
+        results = convert(results)
+
+        data = {
+            key: value / len(results) for key, value in Counter(results).items()
+        }  # Normalize to probabilities
+        return data
+
+    def multistate(
+        self, shots: int = 1, selector: None = None
+    ) -> "np.linalg._linalg.EighResult":
+        """
+        Repeatedly run the task to extract the averaged quantum state.
+          The average is done over nshots and thus is frequentist and converges to
+          exact only in the shots -> infinity limit.
+        Parameters:
+        shots - the number of repetitions of the task
+        selector - an optional callable that takes the output of self.run() and extract
+          the [returned] qubits to be used for the quantum state. If None, all qubits
+          in the simulator are used. Other shan selector = None, the common other pattern is
+           > selector = lambda qubits: qubits
+          for the case where self.run() returns a list of qubits, or
+           > selector = lambda qubit: [qubits]
+          when the output is a single qubit.
+        Returns:
+        np.linalg._linalg.EighResult - the averaged quantum state as a density matrix,
+          represented in its eigenbasis.
```

Comment:
can we make sure we follow the docstring formatting so this renders on the website?